### ISSUE TITLE
`cy.then() timed out after waiting 4000ms. Your callback function returned a promise that never resolved.` on some tests using `axe-core` 4.7.1.

### ISSUE THREAD

**Issue by todd-m-kemp** [component-driven/cypress-axe#160]
I'm trying to upgrade `axe-core` to version 4.7.1 but I'm finding that some of my Cypress a11y tests are now unexpectedly failing with this error:

```
cy.then() timed out after waiting 4000ms.

Your callback function returned a promise that never resolved.

The callback function was:

function (win) {
        if (isEmptyObjectorNull(context)) {
            context = undefined;
        }
        if (isEmptyObjectorNull(options)) {
            options = undefined;
        }
        if (isEmptyObjectorNull(violationCallback)) {
            violationCallback = undefined;
        }
        var _a = options || {}, includedImpacts = _a.includedImpacts, interval = _a.interval, retries = _a.retries, axeOptions = __rest(_a, ["includedImpacts", "interval", "retries"]);
        var remainingRetries = retries || 0;
        function runAxeCheck() {
            return win.axe
                .run(context || win.document, axeOptions)
                .then(function (_a) {
                var violations = _a.violations;
                var results = summarizeResults(includedImpacts, violations);
                if (results.length > 0 && remainingRetries > 0) {
                    remainingRetries--;
                    return new Promise(function (resolve) {
                        setTimeout(resolve, interval || 1000);
                    }).then(runAxeCheck);
                }
                else {
                    return results;
                }
            });
        }
        return runAxeCheck();
    }
```

It seems that the checks that `axe-core` is doing now takes longer than 4 seconds which causes `then` to timeout.

It looks like a workaround is to change the value of Cypress' `defaultCommandTimeout` but having to change this suite-wide or on a case-by-case basis for impacted tests is not a practical solution.

Another option could be to use the `timeout` option for uses of `then` in [checkA11y](https://github.com/component-driven/cypress-axe/blob/682ba11d1d52a00d08a828da25f8a033b67916e0/src/index.ts#L67) but hardcoding it probably isn't a great idea and how long the timeout should be would also be unclear. 🤔

**Comment by MattTreichelYeah** [comment]
I only see this behaviour when the scan finds something wrong so far. I'm not sure what versioning is causing the difference in behaviour because we've recently done some upgrades to major dependencies across the board. We're now on Cypress v12.11.

**Comment by id353** [comment]
Also experiencing this issue with `axe-core` 4.7.2 on a page with no accessibility problems identified.

We've recently migrated from Cypress 8 to 12 and updated `axe-core` and `cypress-axe` as a result. We're now using Cypress 12.13.0 with Electron v160 and `cypress-axe` 1.4.0

Running Cypress locally and clicking through the tests in the UI, everything works well. Once it's pushed to our CI pipelines, the same test times out consistently with the error mentioned above.

We've also tried going back to previous versions of `axe-core` as far back as 4.6.3 and still getting this issue.

**Comment by frodehansen2** [comment]
Any updates on this?

**Comment by todd-m-kemp** [comment]
@martinfer-newsuk Sadly, I have not been able to solve this.

**Comment by frodehansen2** [comment]
@martinfer-newsuk It just started working on the project I had the issue. Now i have the same issue on another project, si I guess i just have to wait until it resolves it self again :)

**Comment by frodehansen2** [comment]
I just discovered that there are some scenarios this error occurs.  Specifically a test where i waited for a named intercept. If i moved the cy.checkA11y() to the next test (which had the same rendered html), everything was fine. Just a tip.

**Comment by todd-m-kemp** [comment]
> Thanks for the replies @frodehansen2 and @todd-m-kemp. The workaround of increasing the `defaultCommandTimeout` solved the errors we were having

That's good to hear! I just don't want to take that approach in our projects. 😂 Patiently awaiting a proper resolution here...

**Comment by marcovdbunt** [comment]
I have the same issue on a page where a webshop-chat is implemented. When chat element is disabled the test is working fine. workaround of increasing the defaultCommandTimeout doesnot solve it
